src/compose.ts

```diff
@@ -226,7 +250,7 @@ function processChanges(composer: Compose) {
   const changes = composer.changes;
   composer.changes = Object.create(null);
 
-  if (!('view' in changes) && !('viewModel' in changes) && ('model' in changes)) {
+  if (!('view' in changes) && !('viewModel' in changes) && ('model' in changes) && composer.activationStrategy !== ActivationStrategy.Replace) {
```

Comment:
Can we do it via something like `resolve/determine ActivationStrategy`? And the exisiting change would be merged into that better, as in we normalize everything in a common handler path?

Reply:
You mean to have a `determineActivationStrategy` hook in the bound VM, like it is done for the routed VMs? 

That actually crossed my mind and it should also be equally easy to implement. Only downside is that every bound VM that needs to be `replace`d should implement this hook as compared to applying a single attribute binding on the instance of `compose`. This might be annoying if the set of bound VMs for a single `compose` instance is considerably big. However, that can also be ameliorated with the help of inheritance.

If you like the trade-off, I am open to change the implementation.

Reply:
Probably we should have both. Like in routing, you can specify the strategy on the route itself, or you can implement the hook in the view model. Does that make sense?

Reply:
That makes perfect sense. It offers more flexibility. 

I will make the change.

Reply:
@bigopon requesting a review for the changes related to the `determineActivationStrategy` hook.